packages/cli/src/commands/virtualdevices/events.ts

```diff
@@ -0,0 +1,182 @@
+		})
+
+		const createEvents = async (_: void, input: DeviceEvent[]): Promise<EventInputOutput> => {
+			const output = await this.client.virtualDevices.createEvents(deviceId, input)
+			return {
+				input,
+				output,
+			}
+		}
+		await inputAndOutputItem<DeviceEvent[], EventInputOutput>(this, {
+			buildTableOutput: (data: EventInputOutput) => buildTableOutput(this.tableGenerator, data),
+		}, createEvents, inputProcessor(() => true, () => this.getInputFromUser(deviceId)))
+	}
+
+	async getInputFromUser(deviceId: string): Promise<DeviceEvent[]> {
+		const attributeName = this.args.name
+		const attributeValue = this.args.value
+		let events: DeviceEvent[] = []
+
+		if (attributeName) {
+			if (attributeValue) {
+				const event = await this.parseDeviceEvent(attributeName, attributeValue, this.args.unit)
+				events = [event]
+			} else {
+				throw new Error('Attribute name specified without attribute value')
+			}
+		}
+		else {
+			const device = await this.client.devices.get(deviceId)
+			const component = await chooseComponent(this, device)
+			const capability = await chooseCapability(this, component)
+			const { attributeName, attribute } = await chooseAttribute(this, capability)
+			const value = await chooseValue(this, attribute, attributeName)
+			const unit = await chooseUnit(this, attribute)
+
+			events = [
+				{
+					component: component.id,
+					capability: capability.id,
+					attribute: attributeName,
+					value: this.convertAttributeValue(attribute.schema.properties.value.type, value),
+					unit,
+				},
+			]
+		}
+		return events
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	convertAttributeValue(attributeType: string | undefined, attributeValue: string): any  {
+		switch (attributeType) {
+			case 'integer':
+				return parseInt(attributeValue)
+			case 'number':
+				return parseFloat(attributeValue)
+			case 'array':
+			case 'object':
+				return JSON.parse(attributeValue)
+			default:
+				return attributeValue
+		}
+	}
+
+	async parseDeviceEvent(attributeName: string, attributeValue: string, unitOfMeasure: string): Promise<DeviceEvent> {
+		const segments = attributeName.split(':')
+		if (segments.length < 2 || segments.length > 3) {
+			throw new Error('Invalid attribute name')
+		}
+
+		const event: DeviceEvent = segments.length === 3 ? {
+			component: segments[0],
+			capability: segments[1],
+			attribute: segments[2],
+			value: attributeValue,
+			unit: unitOfMeasure,
+		} : {
+			component: 'main',
+			capability: segments[0],
+			attribute: segments[1],
+			value: attributeValue,
+			unit: unitOfMeasure,
+		}
+
+		const capability = await this.client.capabilities.get(event.capability, 1)
+		if (!capability || !capability.attributes) {
+			throw new Error(`Capability ${event.capability} not valid`)
+		}
+
+		const attribute = capability.attributes[event.attribute]
+		if (!attribute) {
+			throw new Error(`Attribute ${attributeName} not found in capability ${capability}`)
```

Comment:
We should use this.error() instead in this method.

Reply:
👍